File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.cpp
class Solution {
public:
    int missingMultiple(vector<int>& nums, int k) {
        int maxi = 0;
        unordered_set<int> mp;
        for (int i: nums){
            mp.insert(i);
            maxi = max(maxi , i);
        }

        int i = k;

        while (i <= maxi+k){
            if (i%k == 0 && mp.find(i)==mp.end()) return i;

            i += k;
        }

        return 0;
    }

};

File: 3718-smallest-missing-multiple-of-k/3718-smallest-missing-multiple-of-k.py
from collections import Counter
class Solution:
    def missingMultiple(self, nums: List[int], k: int) -> int:
        mp = Counter(nums)
        n = max(nums)
        i = k
        while i<= n+k:
            if i%k == 0 and i not in mp:
                return i
            i += k
        
